16: Routing I - Login (Part 2)

gabrielizalo opened the pull request:
# Summary

* Second task of Routing: Redux form del login


# Trello Card

https://trello.com/c/uRUucKHu/9-routing-i-login


# Screenshot

<img width="1606" alt="16- routing i - login part 2" src="https://user-images.githubusercontent.com/656353/48228513-5c0e8600-e373-11e8-9b23-8041b4e90224.png">

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- Avances en el Login.

gabrielizalo committed:
Merge branch '16-Routing-I-Login-Part1' into 16-Routing-I-Login-Part2

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- Fixing the name of handle and changing te type of Form.

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- Updating form with Redux Forms

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- Added "Redux Form" and validations to the form.

henryzarza commented on react/react-training/package.json:
```diff
@@ -3,11 +3,13 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "isemail": "^3.2.0",
```
I think that this dependency isn't necessary, please remove it

henryzarza commented on react/react-training/package.json:
```diff
@@ -16,5 +18,8 @@
     "eject": "react-scripts eject"
+  },
+  "devDependencies": {
+    "react-router-dom": "^4.3.1"
```
This isn't `devDependecies` so move it to `dependencies`

henryzarza commented on react/react-training/src/app/components/AppRoutes/index.js:
```diff
@@ -0,0 +1,17 @@
+  render() {
+    return (
+      <Router>
+        <div>
```
Change this for [`Switch`](https://reacttraining.com/react-router/core/api/Switch)

henryzarza commented on react/react-training/src/app/screens/Login/components/SignInForm/components/CustomField/index.js:
```diff
@@ -0,0 +1,15 @@
+
+import styles from '../../styles.scss';
+
+const CustomField = props => {
```
Remember don't create `functional components` as arrow functions

henryzarza commented on react/react-training/src/app/screens/Login/components/SignInForm/components/CustomField/index.js:
```diff
@@ -0,0 +1,15 @@
+import styles from '../../styles.scss';
+
+const CustomField = props => {
+  const { input, type, className, placeholder, meta } = props;
```
In where is the `proptypes`?

henryzarza commented on react/react-training/src/app/screens/Login/components/SignInForm/components/Validation/index.js:
```diff
@@ -0,0 +1,9 @@
+import { validateEmail } from '@utils';
```
I think that this isn't a component, so move it to `utils.js` or `validation.js` file inside of `SignInForm`

henryzarza commented on react/react-training/src/app/screens/Login/components/SignInForm/index.js:
```diff
@@ -0,0 +1,46 @@
+}
+
+SignInForm.propTypes = {
+  handleSubmit: PropTypes.func
```
This is required

henryzarza commented on react/react-training/src/app/screens/Login/components/SignInForm/index.js:
```diff
@@ -0,0 +1,46 @@
+import { required, minLength, isEmail } from './components/Validation';
+import styles from './styles.scss';
+
+class SignInForm extends Component {
```
A good approach is divide this in `dump` and `smart` component.

henryzarza commented on react/react-training/src/app/screens/Login/components/SignInForm/index.js:
```diff
@@ -0,0 +1,46 @@
+};
+
+export default reduxForm({
+  form: 'signIn'
```
This name should be in `constants` file

henryzarza commented on react/react-training/src/app/screens/Login/components/SignInForm/styles.scss:
```diff
@@ -0,0 +1,35 @@
+.error {
+  color: $red;
+  margin: 0 0 10px 10px;
+}
```
Add EOF (End Of File)

henryzarza commented on react/react-training/src/app/screens/Login/components/SignInForm/styles.scss:
```diff
@@ -0,0 +1,35 @@
+}
+
+.error {
+  color: $red;
```
I recommend you change this color or background color because i can't see

henryzarza reviewed (changes requested):
Very good

gabrielizalo committed:
Merge branch '16-Routing-I-Login-Part1' into 16-Routing-I-Login-Part2

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- Fixed dependencies: Removed the unused isEmail and moved from dev
  react-router-dom.

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- Fixed "CustomField" as a functional component. Previously it was
  created as an arrow function.

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- Added the name of the form as a Constant.
- And the propType 'handleSubmit' now is required.

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- The color error was red. Now it's green and verified with 'Color
  Contrast Checker' from https://webaim.org/resources/contrastchecker/
- Renamed the old Green background color to its real name: 'Earls Green'

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- Added the PropTypes to CustomField component
- Fixed the order of the imports into SignInForm component

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- Moved the validation ex-component to validation.js
- Transformed the SignInForm component into a dump and smart component

gabrielizalo commented:
All recommendations are fixed... 👍

gabrielizalo committed:
Merge remote-tracking branch 'origin/master' into 16-Routing-I-Login-Part2

henryzarza commented on react/react-training/src/app/screens/Login/index.js:
```diff
@@ -1,19 +1,15 @@
+  };
+
+  render() {
+    return <SignInForm onSubmit={this.handleSubmit} />;
```
Remember what I said about this component, I do not think this nesting is necessary.

henryzarza commented on react/react-training/src/app/screens/Login/components/SignInForm/layout.js:
```diff
@@ -0,0 +1,44 @@
+import { required, minLength, isEmail } from './validation';
+import styles from './styles.scss';
+
+class Layout extends Component {
```
Change this for `functional component`

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- The Layout from Sign In form now is a functional component

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- Fixed the Login form, removing the nesting form components

gabrielizalo commented:
we are readyyyy... 🥇

wfolini commented on react/react-training/src/app/screens/Login/layout.js:
```diff
@@ -0,0 +1,46 @@
+import { required, minLength, isEmail } from './validation';
+import styles from './styles.scss';
+
+function Layout(props) {
```
Apply the destructuring in the function param
```js
function Layout({ handleSubmit }) {
```

wfolini reviewed (approved):
Great! apply the comment in the next pr
